png_cicp_chunk_proposal.md

```diff
@@ -13,31 +13,33 @@ An existing W3C group note, [BT2100-in-PNG]  specifies an approach which is limi
 * extensible signaling of color space based on H.273
 * does not require the presence of iCCP chunk and embedded ICC profiles
 
+## Non Requirements
+* compatibility with HDR colorspace definitions in any other raster image format (eg. JPEG, AVIF)
+
 ## Strawman approach
-[H.273](https://www.itu.int/rec/T-REC-H.273/en) specifies a controlled vocabulary for the parameterization of
-color space information.
+[H.273](https://www.itu.int/rec/T-REC-H.273/en) specifies a controlled vocabulary for the parameterization of color space information.
 
-Define a cICP chunk that contains the 7 bytes necessary to carry the
-H.273 color space parameters:
+Define a `cICP` chunk that contains the 7 bytes necessary to carry the H.273 color space parameters:
 
-* COLPRIMS, 2 bytes, One of the ColourPrimaries enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
-* TRANSFC, 2 bytes, One of the TransferCharacteristics enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
-* VIDFRNG, 1 byte, Value of the VideoFullRangeFlag specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
+* **COLPRIMS**, 2 bytes, One of the ColourPrimaries enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
+* **TRANSFC**, 2 bytes, One of the TransferCharacteristics enumerated values specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
+* **VIDFRNG**, 1 byte, Value of the VideoFullRangeFlag specified in Rec. ITU-T H.273 | [ISO/IEC 23091-2]
```

Comment:
VideoFullRangeFlag is used in conjunction with MatrixCoefficients in ITU-T H.273, so VIDFRNG does not seem necessary. Is it for limited-range RGB? Thanks.

Reply:
> so VIDFRNG does not seem necessary. 

I think you are right as long as we do not need to carry narrow-range RGB.

@digitaltvguy Any requirement for narrow-range RGB?

Reply:
If you're storing a video frame that has details in the sub-blacks or super-whites (e.g. the output from a live camera), you may want to store it as narrow range, otherwise there will be clipping which will affect downstream filters.

Reply:
PNG was never intended for narrow range... so I am not convinced it is a good idea to start now.

Reply:
I have not seen narrow-range data stored in PNG images.

On the other hand, single frames from narrow-range video would be easier to store in PNG if a narrow-to-full rescaling was not required.

Reply:
Please include VideoFullRangeFlag 
VideoFullRangeFlag can also be used for RGB images, so it is relevant for some use case.
There are several narrow range RGB files in use as most apps don't support YCC files.

Reply:
Why are the fields 2 bytes? H.273 supports only 1 byte.

Reply:
Per the 2021-02-24 meeting:
* there is a desire to support narrow range images, RGB and YUV
* `VideoFullRangeFlag` can signal narrow range for both RGB (the identity matrix is used) and YUV
* all four fields shall be present in a given cICP chunk

Reply:
I made the change here *however* because of @svgeesus comments about YUV, I am not marking this resolved.  @palemieux